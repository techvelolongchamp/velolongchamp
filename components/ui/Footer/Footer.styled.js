import styled from 'styled-components'

export const Wrapper = styled.footer`
  background: ${({ theme }) => theme.colors.offWhite};
  border-top: 1px solid ${({ theme }) => theme.colors.offWhite};
  color: ${({ theme }) => theme.colors.lightGrey};
  padding: ${({ theme }) => theme.spacing.s} 0
    ${({ theme }) => theme.spacing.xs};
  text-align: center;
  a {
    color: ${({ theme }) => theme.colors.lightGrey};
    &:hover {
      text-decoration: underline;
    }
  }
`
