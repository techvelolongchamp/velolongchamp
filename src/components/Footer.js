import React from 'react'
import { Link } from 'gatsby'
import styled from 'styled-components'

const Wrapper = styled.footer`
  background: ${({ theme }) => theme.colors.offWhite};
  border-top: 1px solid ${({ theme }) => theme.colors.offWhite};
  color: ${({ theme }) => theme.colors.lightGrey};
  padding: ${({ theme }) => theme.spacing.s} 0
    ${({ theme }) => theme.spacing.xs};
  text-align: center;
  a {
    color: ${({ theme }) => theme.colors.lightGrey};
    &:hover {
      text-decoration: underline;
    }
  }
`

const Footeri18n = {
  fr: {
    allRights: 'Tous droits réservés',
    legal: 'Mentions légales',
  },
  en: {
    allRights: 'All rights reserved',
    legal: 'Legal Notice',
  },
}

const Footer = ({ lng }) => {
  const trad = Footeri18n[lng]
  const now = new Date()
  const year = now.getFullYear()
  return (
    <Wrapper>
      <p>
        ©Copyright 2019{year > 2019 && ` - ${year}`} - Vélo Longchamp -{' '}
        {trad.allRights}
      </p>
      <p>-</p>
      <p>
        <Link to="/legal">{trad.legal}</Link>
      </p>
    </Wrapper>
  )
}

export default Footer
