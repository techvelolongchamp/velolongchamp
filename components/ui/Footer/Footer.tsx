import React from 'react'
import Link from 'next/link'
import { useIntl } from 'react-intl'

import { Wrapper } from './Footer.styled'

const Footer = () => {
  const { formatMessage } = useIntl()
  const now = new Date()
  const year = now.getFullYear()
  return (
    <Wrapper>
      <p>
        ©Copyright 2019{year > 2019 && ` - ${year}`} - Vélo Longchamp -{' '}
        {formatMessage({ id: 'footer.allRights' })}
      </p>
      <p>-</p>
      <p>
        <Link href="/legal">{formatMessage({ id: 'footer.legal' })}</Link>
      </p>
    </Wrapper>
  )
}

export default Footer
