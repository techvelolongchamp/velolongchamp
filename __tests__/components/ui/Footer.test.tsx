import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import Footer from '../../../components/ui/Footer/Footer'

describe('Footer unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })
})
