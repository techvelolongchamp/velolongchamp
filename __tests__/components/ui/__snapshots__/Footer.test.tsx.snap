// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer unit tests should render correctly 1`] = `
.c0 {
  background: #f7f7F7;
  border-top: 1px solid #f7f7F7;
  color: #999999;
  padding: 10px 0 5px;
  text-align: center;
}

.c0 a {
  color: #999999;
}

.c0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <footer
    class="c0"
  >
    <p>
      ©Copyright 2019
       - 2023
       - Vélo Longchamp -
       
      Tous droits réservés
    </p>
    <p>
      -
    </p>
    <p
      id="legal-link"
    >
      <a
        href="/legal"
      >
        Mentions légales
      </a>
    </p>
  </footer>
</div>
`;
